jmetal-core/src/main/java/org/uma/jmetal/qualityindicator/impl/GeneralizedSpread.java

```diff
@@ -111,7 +111,10 @@ public double generalizedSpread(Front front, Front referenceFront) {
             dmean);
       }
 
-      return (dExtrems + mean) / (dExtrems + (numberOfPoints*dmean));
+      double denominator=(dExtrems + (numberOfPoints*dmean));
+      if(denominator==0) return -1;
+      
+      return (dExtrems + mean) / denominator;
```

Comment:
The semantic changes here:
- before, a zero denominator generates an exception
- after, a zero denominator returns a -1 result

-1 are not idiomatic in Java to indicate something went wrong. Unless this value is motivated by the algorithm, it should actually generate an exception. And the old code was precisely doing that. We may catch it to return a contextualised error message, but returning -1 does not seem to be the right choice to me.

By the way, as we take the hypothesis that `numberOfPoints` is strictly positive in line 101 above, `dmean` appears to be strictly positive too. And since `dExtrems` is build similarly, the same hypothesis can be taken. Thus, the denominator cannot be zero unless something went wrong, like having zero points or zero objectives, which is clearly an issue and thus should generate an exception, not a mere default value.

Overall, it seems we do a late check while many earlier checks are not done (front emptiness, `numberOfPoints` > 0, etc.)